isis/tests/FunctionalTestsOverlapstats.cpp

```diff
@@ -0,0 +1,259 @@
+  EXPECT_EQ( (double) group.findKeyword("ThicknessAverage"), 0.29119530568336);
+  EXPECT_EQ( (double) group.findKeyword("ThicknessStandardDeviation"), 0.21190468305603);
+  EXPECT_EQ( (double) group.findKeyword("ThicknessVariance"), 0.044903594701078);
+  EXPECT_EQ( (double) group.findKeyword("AreaMinimum"), 49665309599.111);
+  EXPECT_EQ( (double) group.findKeyword("AreaMaximum"), 125515198928.3);
+  EXPECT_EQ( (double) group.findKeyword("AreaAverage"), 87590254263.708);
+  EXPECT_EQ( (double) group.findKeyword("AreaStandardDeviation"), 53633971096.922);
+  EXPECT_EQ( (double) group.findKeyword("AreaVariance"), 2.87660285562549e+21);
+  EXPECT_EQ( (double) group.findKeyword("ImageStackMinimum"), 2.0);
+  EXPECT_EQ( (double) group.findKeyword("ImageStackMaximum"), 3.0);
+  EXPECT_EQ( (double) group.findKeyword("ImageStackAverage"), 2.5);
+  EXPECT_EQ( (double) group.findKeyword("ImageStackStandardDeviation"), 0.70710678118655);
+  EXPECT_EQ( (double) group.findKeyword("ImageStackVariance"), 0.5);
+  EXPECT_EQ( (int) group.findKeyword("PolygonCount"), 3);
+}
+
+
+TEST_F(ThreeImageNetwork, FunctionalTestOverlapstatsFull) {
+  // write points so that cube2 is a full overlap of cube1
+  lonLatPts = new geos::geom::CoordinateArraySequence();
+  lonLatPts->add(geos::geom::Coordinate(31, 1));
+  lonLatPts->add(geos::geom::Coordinate(31, 9));
+  lonLatPts->add(geos::geom::Coordinate(34, 9));
+  lonLatPts->add(geos::geom::Coordinate(34, 1));
+  lonLatPts->add(geos::geom::Coordinate(31, 1));
+
+  polys = new std::vector<geos::geom::Geometry *>;
+  poly = globalFactory->createPolygon(globalFactory->createLinearRing(lonLatPts), nullptr);
+  polys->push_back(poly->clone());
+  multiPoly = globalFactory->createMultiPolygon(polys);
+
+  geos::io::WKTWriter *wkt = new geos::io::WKTWriter();
+
+  std::string polyStr = wkt->write(multiPoly);
+  int polyStrSize = polyStr.size();
+  std::istringstream polyStream(polyStr);
+
+  Blob pvlBlob("Footprint", "Polygon");
+  Pvl pvl;
+  PvlObject polyObject = PvlObject("Polygon");
+  polyObject.addKeyword(PvlKeyword("Name", "Footprint"));
+  polyObject.addKeyword(PvlKeyword("StartByte", "1"));
+  polyObject.addKeyword(PvlKeyword("Bytes", toString(polyStrSize)));
+  pvl.addObject(polyObject);
+
+  pvlBlob.Read(pvl, polyStream);
+  cube2->write(pvlBlob);
+  cube2->reopen("rw");
+
+  FileList cubes;
+  cubes.append(FileName(cube1->fileName()));
+  cubes.append(FileName(cube2->fileName()));
+  cube1->close();
+  cube2->close();
+
+  QString cubeListPath = tempDir.path() + "/cubes.lis";
+  QString overlapsPath = tempDir.path() + "/overlaps.lis";
+  cubes.write(cubeListPath);
+
+  // get overlaps for overlapstats input
+  QString FIO_APP_XML = FileName("$ISISROOT/bin/xml/findimageoverlaps.xml").expanded();
+  QVector<QString> fio_args = { "OVERLAPLIST=" + overlapsPath };
+  UserInterface fioOptions(FIO_APP_XML, fio_args);
+  findimageoverlaps(cubes, fioOptions);
```

Comment:
I would try to get these calls out of here as it's seems cleaner to not make tests depend on multiple apps, assuming replacing it with a pre-baked overlap file isn't too much of a hassle.